R/utils.R

```diff
@@ -163,6 +163,46 @@ proceed <- function(default = TRUE) {
   ask("Do you want to proceed?", default = default)
 }
 
+menu <- function(choices, title, default = 1L) {
+  testing <- getOption("renv.menu.choice", integer())
+  if (length(testing)) {
+    selected <- testing[[1]]
+    options(renv.menu.choice = testing[-1])
+  } else if (is_testing()) {
+    selected <- default
+  } else {
+    selected <- NULL
+  }
+
+  if (!is.null(selected)) {
+    writef(c(
+      title,
+      "",
+      paste0(seq_along(choices), ": ", choices),
+      "",
+      paste0("Selection: ", selected),
+      ""
+    ))
+    return(names(choices)[selected])
+  }
+
+  if (!interactive()) {
+    writef(c("Not interactive. Will:", choices[[default]]))
+    return(default)
+  }
+
+  tryCatch(
+    idx <- utils::menu(choices, paste(title, collapse = "\n"), graphics = FALSE),
```

Comment:
```suggestion
  idx <- tryCatch(
    utils::menu(choices, paste(title, collapse = "\n"), graphics = FALSE),
```

since `tryCatch()` is always producing a usable value?

Reply:
Good catch. I'll fix this by hand.